src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -2713,5 +2715,40 @@ private static ProjectItem GetOneItem(string content)
             Assert.Equal(1, items.Count);
             return items[0];
         }
+
+        /// <summary>
+        /// Item metadata "Filename" should not depends on platform specific slashes.
+        /// </summary>
+        [Fact]
+        public void FileNameMetadataEvaluationShouldNotDependsFromPlatformSpecificSlashes()
+        {
+            var projectA = Path.Combine(ObjectModelHelpers.TempProjectDir, "a.proj");
+
+            const string contentsA = @"<?xml version=`1.0` encoding=`utf-8`?>
+<Project ToolsVersion=`msbuilddefaulttoolsversion` DefaultTargets=`Validate` xmlns=`msbuildnamespace`>
+  <ItemGroup>
+    <A Include=`A\B\C\D.cs` />
+    <B Include=`@(A->'%(RootDir)%(Directory)%(Filename)_test.ext')` />
+  </ItemGroup>
+</Project>";
+
+            try
+            {
+                File.WriteAllText(projectA, ObjectModelHelpers.CleanupFileContents(contentsA));
+                var projectCollection = new ProjectCollection();
+                var project = new Project(projectA, new Dictionary<string, string>(), null, projectCollection);
+                var buildManager = BuildManager.DefaultBuildManager;
+                var projectInstance = buildManager.GetProjectInstanceForBuild(project);
+                var itemB = projectInstance.Items.Single(i => i.ItemType == "B").EvaluatedInclude.Replace("/", "\\");
+                itemB.ShouldNotContain("A\\B\\C\\A\\B\\C");
```

Comment:
Please assert the exact expected value, not what it shouldn't contain, since there's only one correct value, but many incorrect ones.